connect/api/src/main/java/org/apache/kafka/connect/data/ConnectSchema.java

```diff
@@ -291,7 +291,7 @@ public boolean equals(Object o) {
                 Objects.equals(name, schema.name) &&
                 Objects.equals(doc, schema.doc) &&
                 Objects.equals(type, schema.type) &&
-                Objects.equals(defaultValue, schema.defaultValue) &&
+                Objects.deepEquals(defaultValue, schema.defaultValue) &&
```

Comment:
Note this falls back to the regular `equals()` for non-array types. One might consider to either call `equals()` or `deepEquals()` based on the schema's type, but it felt like a potential bug lurking if future types are going to be added:

    (((type == ARRAY || type == BYTES) && Objects.deepEquals(defaultValue, schema.defaultValue)) ||
        Objects.equals(defaultValue, schema.defaultValue)) && ...

Happy to change it if you think it's preferable.

Reply:
My only reservation is that `Objects.deepEquals` checks quite a few conditions before it falls back to the regular `equals()`. However, I agree that simpler is probably better, and in the grand scheme of things (no pun intended) the better approach.